perl/lib/Lingy/nREPL.pm

```diff
@@ -53,20 +53,52 @@ sub new {
 #------------------------------------------------------------------------------
 # nREPL server op codes handlers:
 #------------------------------------------------------------------------------
+package StreamedOutput;
+
+sub TIEHANDLE {
+    my ( $class, $send_response, $output_type ) = @_;
+    bless {
+        send_response => $send_response,
+        output_type => $output_type
+    }, $class;
+}
+
+sub PRINT {
+    my ( $self, @args ) = @_;
+    my $output_type = $self->{output_type} // 'out';
+    $self->{send_response}->( { $output_type => join '', @args } );
+}
+
+sub PRINTF {
+    my ( $self, $format, @args ) = @_;
+    $self->PRINT( sprintf $format, @args );
+}
+
+package Lingy::nREPL;
+
 sub op_eval {
     my ($self) = @_;
 
+    my $code = $self->{request}->{code};
+
+    local *STDOUT;
+    tie *STDOUT, 'StreamedOutput', sub { $self->send_response(@_) }, 'out';
+
+    local *STDERR;
+    tie *STDERR, 'StreamedOutput', sub { $self->send_response(@_) }, 'err';
+
     my $result;
     eval {
-        my $code = $self->{request}{code};
         my @results = $self->{repl}->reps($code);
         $result = $results[-1];
+        $self->send_response( { value => $result } );
```

Comment:
`@results` is an array of the result for each form.
I'm not sure if it can be empty or not.
We might want to say
```
 $self->send_response( { value => $result } )
     if @response;
```

Reply:
And if we do that we probably shouldn't send `done` if we don't send a response.

Reply:
Also for single pair `send` I've been formatting like:
```
$self->send_response({value => $result});
```
Let's keep the style consistent, whatever it is.

Reply:
> And if we do that we probably shouldn't send `done` if we don't send a response.

I think we should. Calva needs to know when the server is done with the request. I have never seen a non-response with the Java nREPL server, unless the evaluation goes wrong. We should consider sending an `err` message if this happens.